static/app/components/devtoolbar/types.ts

```diff
@@ -27,9 +27,12 @@ type QueryKeyEndpointOptions<
   query?: Query;
 };
 
+// Prefix the key with a namespace, to avoid key collisions with other tanstack/query
+// cache requests that imported sentry modules make within the toolbar scope.
 export type ApiQueryKey =
-  | readonly [url: string]
+  | readonly ['io.sentry.toolbar', url: string]
   | readonly [
+      'io.sentry.toolbar',
```

Comment:
This is annoying, but for now it's the only solution i've got.

What's happening is we're importing components from sentry. Those components can fetch data using `static/app/utils/queryClient.tsx` like usual. The problem with calling `queryClient.tsx` is that it depends on ConfigStore and a bunch of internals. So, long term the toolbar shouldn't depend on those (by default shouldn't), which is why we have `static/app/components/devtoolbar/hooks/useApiEndpoint.tsx` for the toolbar, it's much simpler and skips a bunch of stuff, but should be ok for what we're doing.

The problem is that sentry components (that are imported into the toolbar), and toolbar components must share the same query cache. They could have the same cache-key in there, with different results cached. This is not good.

So what i've done is added a cache-busting namespace string to the start of all new keys that the toolbar uses. Fixed!